### ISSUE TITLE
adding Skip flag in pom plugin configuration

### ISSUE THREAD

**Issue by britishdan** [davidB/scala-maven-plugin#198]
Hello!
I'm a noob at Maven plugins so excuse me if I'm not making much sense.

We have a scenario where we would like to skip the execution() methods of the scala-maven-plugin.
We have seen that there is a skip flag in other native plugins which is checked in execution().
It seems to be a common practise. Surefire does it for example: https://github.com/apache/maven-surefire/blob/4d3673c42e2af75c22f3abbaa2a0237a5c25e0ce/maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java#L153.

Basically we want a way to skip the scala-maven-plugin entirely.
WDYT?

**Comment by davidB** [comment]
Hi,

BTH I don't know other plugin than maven-surefire-plugin, with a skip property. Anyway, having a skip is valid when the plugin is doing one things (goal) like test. But the scala-maven-plugin as lot of goals.

IIRC, with maven's profile you can simulate a "skip" property (but more verbose).

What is your use case ?

PS: PR are welcomes.

**Comment by britishdan** [comment]
Thanks for the response! We did end up using maven profiles to simulate the skip.

As you said, there are multiple goals and we see that only one of them (ScalaDocJarMojo) has a skip property. Is there a reason the others do not?

Our use case is running Integration Tests in Docker. The plan is to let maven run as usual, but perform the **verify** stage in Docker. The problem is that running **mvn verify** in the Docker causes all the previous steps to unnecessarily run again. This causes the build to take a very long time. To solve it, we've added a skip in the pom to all the plugins when running in the Docker. This worked for all the plugins except the scala plugin.

We ended up solving it by using maven profiles.

**Comment by ehiggs** [comment]
>We have a scenario where we would like to skip the execution() methods of the scala-maven-plugin.
To skip scaladoc generation you need to use `-Dskip=true`. This should almost certainly be changed to `maven.scaladoc.skip`.

>BTH I don't know other plugin than maven-surefire-plugin, with a skip property. Anyway, having a skip is valid when the plugin is doing one things (goal) like test. But the scala-maven-plugin as lot of goals.

`maven-javadoc-plugin` has `-Dmaven.javadoc.skip`.

**Comment by reggert** [comment]
Also note that the `maven-compiler-plugin`, which is the closest to this one in terms of purpose, also supports a "skip" option. In this case, it shows up as `<skipMain>` in the POM or `maven.main.skip` on the command line. The `testCompile` goal correspondingly supports `<skip>` and `maven.test.skip`.

One major use case is when you want to force a reactor build without actually running the compiler, such as if you're trying to run `dependency:sources` on a multi-module project in order to download source attachments, which doesn't work correctly if you just run `mvn dependency:sources` on its own (it tries to resolve the inter-module dependencies without using the reactor).

**Comment by slandelle** [comment]
IMHO, this would be a good first contribution.

**Comment by slandelle** [comment]
No activity, closing. Will reopen in case someone is willing to contribute.

**Comment by timoninmaxim** [comment]
@slandelle hi! I'd like to implement the "maven.scaladoc.skip" flag. It will make the configuration of our project a little bit transparent. Should I create a new issue for that?

**Comment by nigredo-tori** [comment]
It is sad that this went nowhere. `maven.main.skip` and `maven.test.skip` support (or some plugin-specific equivalent) would simplify Bloop import process which is a part of the Metals workflow. As it stands, to properly import a multi-module project with protobuf generation I have to do something like this:
```bash
$ mvn generate-sources compile dependency:sources ch.epfl.scala:maven-bloop_2.13:$BLOOP_VERSION:bloopInstall
```

which means that after each build change I have to compile the project before I can start working on it. As you can imagine, this gets old fast, especially if the change happens to break existing code - in which case I have to fix stuff with Metals disabled.

**Comment by slandelle** [comment]
@timoninmaxim @nigredo-tori Feel free to provide a PR targeting this ticket and I'll reopen it.

@nigredo-tori Regarding interactions with Bloop, I regret that the developers there chose to have their maven-bloop extend classes of this scala-maven-plugin instead of contributing upstream. It makes things very brittle and coordination impossible.
